daprdocs/content/en/developing-applications/develop-components/pluggable-components/develop-pluggable.md

```diff
@@ -105,6 +105,167 @@ After generating the above state store example's service scaffolding code using
 
 This concrete implementation and auxiliary code are the **core** of your pluggable component. They define how your component behaves when handling gRPC requests from Dapr.
 
+## Return semantic state store errors
+
+In order to meet concurrency requirements, your state store pluggable components need error mapping, from gRPC returned errors to semantic errors. Each of the three Dapr state store errors below maps to a gRPC status code. 
+
+| Error                    | gRPC error code                 | 
+| ------------------------ | ------------------------------- | 
+| ETag Mismatch            | `codes.FailedPrecondition`          | 
+| ETag Invalid             | `codes.InvalidArgument`           | 
+| Bulk Delete Row Mismatch | `codes.Internal` | 
+
+Learn more about concurrency requirements in the [State Management overview]({{< ref "state-management-overview.md#concurrency" >}}).
+
+The following examples demonstrate how to return an error in your own pluggable component, changing the messages to suit your needs.
+
+{{< tabs ".NET" "Java" "Go" >}}
+ <!-- .NET -->
+{{% codetab %}}
+
+> **Important:** In order to use .NET for error mapping, first install the [`Google.Api.CommonProtos` NuGet package](https://www.nuget.org/packages/Google.Api.CommonProtos/).
+
+**Etag Mismatch**
+
+```csharp
+var badRequest = new BadRequest();
+var des = "The ETag field must only contain alphanumeric characters";
```

Comment:
Is this error message correct? It is the same as the ETag Invalid.

```suggestion
var des = "The ETag field provided does not match the one in the store";
```

Reply:
good catch